### PR TITLE
add Helix send chat message

### PR DESCRIPTION
**Pull request by zunderscore**
Type: Feature

Fixes: #541

Adds new Helix send chat message endpoint

### CODE REVIEW

File: packages/api/src/endpoints/chat/HelixChatApi.ts [hunk 1]
```diff
@@ -251,6 +256,35 @@ export class HelixChatApi extends BaseApi {
 		return new HelixPrivilegedChatSettings(result.data[0]);
 	}
 
+	/**
+	 * Sends a chat message to a broadcaster's chat.
+	 *
+	 * This uses the token of the broadcaster by default.
+	 * If you want to execute this in the context of another user (who has to be moderator of the channel)
```
Comment (d-fischer) [inline packages/api/src/endpoints/chat/HelixChatApi.ts:263]: The user does not need to be a moderator.

Additionally, there should probably be a way to send explicitly via the app token capabilities - this probably needs to be a separate method, as the sender/scope logic doesn't work in a `.withoutUser` context.

File: packages/api/src/endpoints/chat/HelixChatApi.ts [hunk 2]
```diff
@@ -251,6 +256,35 @@ export class HelixChatApi extends BaseApi {
...
+	 * you can do so using [user context overrides](/docs/auth/concepts/context-switching).
+	 *
+	 * @param broadcaster The broadcaster the chat belongs to.
+	 * @param message The message to send.
+	 */
+	async sendChatMessage(
+		broadcaster: UserIdResolvable,
+		message: HelixSendChatMessageParams,
```
Comment (d-fischer) [inline packages/api/src/endpoints/chat/HelixChatApi.ts:271]: For simplicity in the common case, as well as for being in line with the `ChatClient` a bit more, I'd prefer if the message text was a separate parameter to the method. The params object should still stay, even though it (currently) only has one property.

**Review by d-fischer — CHANGES_REQUESTED** [review]

File: packages/api/src/endpoints/chat/HelixChatApi.ts (hunk 1 above)
Comment (zunderscore) [reply packages/api/src/endpoints/chat/HelixChatApi.ts]: @d-fischer I've updated the JSDoc. I don't know how you want to implement the separate app token method, so you'd either need to provide an example of an existing implementation of this, or add it in yourself. You have write access to my branch for convenience.

File: packages/api/src/endpoints/chat/HelixChatApi.ts [hunk 3]
```diff
@@ -251,6 +256,39 @@ export class HelixChatApi extends BaseApi {
...
+	 * @expandParams
+	 *
+	 * @param broadcaster The broadcaster the chat belongs to.
+	 * @param message The message to send.
+	 * @param params
+	 */
+	async sendChatMessage(
+		broadcaster: UserIdResolvable,
+		message: string,
+		params: HelixSendChatMessageParams,
```
Comment (d-fischer) [inline packages/api/src/endpoints/chat/HelixChatApi.ts:275]: This can now be optional.

**Review by d-fischer — CHANGES_REQUESTED** [review]

**Review by d-fischer — APPROVED** [review]

**Pull request merged**